src/object/srv_obj.c

```diff
@@ -2316,7 +2319,27 @@ obj_verify_bio_csum(crt_rpc_t *rpc, daos_iod_t *iods,
 		daos_sgl_fini(&sgl, false);
 
 		if (rc != 0) {
-			D_ERROR("Verify failed: %d\n", rc);
+			if (iod->iod_type == DAOS_IOD_SINGLE) {
+				D_ERROR("Data Verification failed (object: "
+				DF_OID" dkey: %s, akey: %s): %d\n",
+					DP_OID(orw->orw_oid.id_pub),
+					(char *)orw->orw_dkey.iov_buf,
+					(char *)iod->iod_name.iov_buf, rc);
+			}
+			if (iod->iod_type == DAOS_IOD_ARRAY) {
+				D_ERROR("Data Verification failed (object: "
+				DF_OID" dkey: %s, akey: %s, "
+				"extent: [%lu-%lu]): %d\n",
+					DP_OID(orw->orw_oid.id_pub),
+					(char *)orw->orw_dkey.iov_buf,
+					(char *)iod->iod_name.iov_buf,
```

Comment:
In general, we should avoid printing user data (keys) into the log file.   But even if we do print them, they should be done with %*s, (int)iov->iov_len, (char *)iov->iov_buf so we limit the print to only the buffer in case there is no null byte.

Reply:
good call. removing the keys